### DISCUSSION TITLE
How to disable asset embedding?

### DISCUSSION THREAD

**Discussion by XX** [tauri-apps/tauri#12061] [Q&A] [answered]
I want to change the contents of the `dist` directory while the tauri-app is running, add and remove web apps there. But by default, tauri embeds the contents of this directory into the executable file when compiling. How can I change this behavior?

**Comment by FabianLars** [comment, marked as answer]
If i understand it right, you're basically asking for https://github.com/tauri-apps/tauri/issues/5302. As you can see, this is not supported yet.

What you could do is point frontendDist to an empty dir, or a url, and then host your frontend files via your own localhost server.

**Comment by XX** [comment]
@FabianLars Is it possible to do this by custom [Assets](https://docs.rs/tauri/latest/tauri/trait.Assets.html) implementation?

**Reply by FabianLars** [reply]
Probably yes, never given that trait much thoughts myself tbh.

**Reply by XX** [reply]
I wrote my own implementation of the `Assets` and set it into the generated context. It works!

**Reply by FabianLars** [reply]
Awesome 👏

**Reply by WSH032** [reply]
@XX, can you share some code examples with me? ❤ I am very interested in this.

I am the maintainer of [pytauri](https://github.com/WSH032/pytauri), and I am working on implementing Python bindings for Tauri.

To lower the learning curve for Python developers, I want to precompile Tauri into a dynamic library (called `Wheel` in Python) and then call this dynamic library in Python.

However, providing frontend assets has always been a problem for me. Currently, I can only serve assets through a local server ([ref](https://github.com/pytauri/example-nicegui-app/blob/52f8a1265111b6be6db4814588599e8e3370ffb9/tauri.conf.json#L8), [ref](https://github.com/pytauri/example-nicegui-app/blob/52f8a1265111b6be6db4814588599e8e3370ffb9/python/nicegui_app/__init__.py#L96)).

But this is far from ideal. I believe it goes against Tauri's `no localhost server` philosophy, and the local server **significantly increases memory consumption** (in my tests, using [uvicorn](https://github.com/encode/uvicorn) caused the entire program to consume 160MB of memory).

If I could use a custom `Assets` implementation, I could serve assets directly from the file system without relying on a local server.
